lang/c++/impl/ValidSchema.cc

```diff
@@ -145,49 +140,49 @@ ValidSchema::toFlatList(std::ostream &os) const
  * in UTF-8 format. Note that this method is not responsible for validating
  * the schema.
  */
-string ValidSchema::compactSchema(const string& schema) {
-    bool insideQuote = false;
-    size_t newPos = 0;
-    string data(schema.data());
-
-    for (size_t currentPos = 0; currentPos < schema.size(); currentPos++) {
-        if (!insideQuote && std::isspace(data[currentPos])) {
-            // Skip the white spaces outside quotes.
-            continue;
-        }
+    string ValidSchema::compactSchema(const string &schema) {
+        auto insideQuote = false;
```

Comment:
changes like this to auto don't really seem to add value?

Reply:
Scott Mayer's advice (in Effective Modern C++) is to use `auto` unless, the default detection is something you don't want.

Reply:
That is fair.  I'm not sure I buy all of his arguments of Meyer's, I think it is highly dependent assumed tooling..  I think when refactoring it doesn't really provide value to replace the types, and can introduce bugs (as noted on the integer narrowing).  It seems like a good practice for new code?